src/dvc_data/index/view.py

```diff
@@ -114,6 +114,9 @@ def node_factory(_, key, children, entry=None):
 
             return key
 
+        self._index._ensure_loaded(  # pylint: disable=protected-access
+            root_key
+        )
```

Comment:
Dir loading should be lazier for views (we may not need to load the entire directory tree depending on the view)

We should be re-using the node factory based traversal and view._load_dir_keys from view._iteritems() here (but for ls we would yield info entries instead of key/value pairs)